iXBRLViewerPlugin/viewer/src/js/report.js

```diff
@@ -69,6 +69,22 @@ iXBRLReport.prototype._initialize = function () {
     }
 }
 
+iXBRLReport.prototype.isCalculationContributingConcept = function(c) {
+    if (!this.data.rels?.calc) return false;
+    return Boolean(Object.values(this.data.rels.calc).some(calculations => {
+        return Object.values(calculations).some(contributingItems => {
+            return contributingItems.some(i => i.t === c);
```

Comment:
In thinking about startup time, this appears to be O(facts*contributors).  It seems like we could compute a set of contributors so that this method can just be `this.contributingItems.has(c)`, and similarly for summations.  We could change fact.isCalcSum/item to be methods rather than fields (or to be lazily computed fields), and then change this to be a lazily computed set of contributors to avoid affecting startup at all.